shop/src/main/java/ro/msg/learning/shop/controller/ProductController.java

```diff
@@ -0,0 +1,144 @@
+package ro.msg.learning.shop.controller;
+
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.dto.ProductCombinedDto;
+import ro.msg.learning.shop.dto.SupplierDto;
+
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.service.exceptions.ProductException;
+import ro.msg.learning.shop.utils.Status;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+@RestController
+public class ProductController {
+    private final ProductService productService;
+
+    public ProductController(ProductService productService){
+        this.productService=productService;
+    }
+     @GetMapping("/suppliers")
+    @ResponseBody
+    public List<SupplierDto> getAllSupliers(){
```

Comment:
Do not combine the endpoints for all the entities on the same Controller. Create a controller for each entity.

Reason: Single Responsability Principle